parser/parser.go

```diff
@@ -194,7 +204,7 @@ func (p *Parser) Parse() (*gonginx.Config, error) {
 }
 
 // ParseBlock parse a block statement
-func (p *Parser) parseBlock(inBlock bool) (*gonginx.Block, error) {
+func (p *Parser) parseBlock(inBlock bool, isSkipValidDirective bool) (*gonginx.Block, error) {
```

Comment:
@mofantor  I like the solution here, do you think you can pass the all options (for the future when we need to check more options recursively) here?

Reply:
probably not, and if we need to check more options ,we should use options collection to pass and validate( for future)

Reply:
@tufanbarisyildirim I have just finished two tasks in the project and I am ready to submit PR. If you are interested, you can come to the project and have a look。

- move any context wrapper into their own file (remove from parser)
- link the parent directive to any directive for easier manipulation

Reply:
@tufanbarisyildirim project : https://github.com/mofantor/gonginx.git

Reply:
Yeah I already saw them, would love to see they are merging here! I will allocate some more time to go over today.